src/ast/query.rs

```diff
@@ -695,6 +695,13 @@ pub enum TableFactor {
         expr: Expr,
         alias: Option<TableAlias>,
     },
+    /// `e.g. LATERAL FLATTEN(<args>)[ AS <alias> ]`
+    Function {
```

Comment:
In general it would probably be good to consolidate the `TableFunction` variant with `Function` but I think this is a pragmatic way to add the functionality.

Thank you @lustefaniak